lib/dbs/firebase.ts

```diff
@@ -91,6 +103,20 @@ export async function deleteUser({ context, user, sub }: SessionProps) {
     await deleteDoc(ref);
 }
 
+export async function deleteStoreUsers({ context, sub }: SessionProps) {
+    const contextString = context ?? sub;
+    const storeHash = contextString?.split('/')[1] || '';
+    const storeUsersRef = query(collection(db, 'storeUsers'), where('storeHash', '==', storeHash));
+    const storeUsers = await getDocs(storeUsersRef);
+    const batch = writeBatch(db);
```

Comment:
@bc-zachary @christensenep - we likely should not be removing collections like this. There's a couple of thoughts here. First and foremost, we always have to be careful when dealing with databases, especially with bulk transactions. Google actually goes as so far to explicitly say we should not be using this pattern with web apps as it has major security implications - they even exclude the Firebase v9 sample code (which would actually look somewhat similar to this) - https://firebase.google.com/docs/firestore/manage-data/delete-data#web.

Second, security implications aside, we likely should have some basic error handling. We might also want to consider some additional batch config/ handling (e.g. batch size). As it is right now, we're processing everything in one batch.  This might be fine for a really small app, but could become problematic on a larger app.

Reply:
I don't actually think we *are* deleting a collection here, are we?

Reply:
It depends. theoretically, there could be one store in which case it will delete the entire collection. I think, though, the larger issue is the sort of bulk removal from an unsecured front-end JS SDK. Specifically, since this action is being taken from an accessible address, e.g. `storeHash.mybigcommerce.com/manage/app/APPID/api/uninstall`, we currently have no way of authenticating that the action is being taken from an authorized user, save for the JWT, but that's the same exact JWT that gets sent on `load` - meaning anyone who can load the app, or anyone with the JWT, could theoretically remove all of the users.

Reply:
Ok, so to sum up:

We have indeed found a vulnerabilty, not with the sample app specifically, but with our app callbacks generally, in which the jwt can be used interchangably with any endpoint (load/uninstall/install). Most notably this could be used to do weird stuff with the uninstall endpoints, but in theory accessing any of them with the wrong jwt is bad (and this would apply to any future endpoints we might add). This is captured [here](https://bigcommercecloud.atlassian.net/browse/APPEX-289)

Regarding the actual implementation of the uninstall callback here, though, are we on the same page that this implementation is fine? One thing I do think we may need to add is handling batches of over 500 users (I believe batches of 500 operations is the max for firebase). But I don't think there's anything fundamentally wrong/insecure about performing these deletions here. Notably, this is actually executed on the backend, not the frontend. Any reason not to revert the revert (possibly adding the 500 count case, although I don't personally think we need to wait on that).

Reply:
The only caveat here is that we will need to update the code to check that the JWT was called with uninstall - otherwise, we're merging in the original vulnerability. The 500 batch fix would be nice

Reply:
Oh that latter part I don't think we can do yet, need a change on the App Registry side of things (in the ticket above), the JWT simply lacks any information that says which endpoint its meant to be used with. I think we can probably tackle that once the above ticket is done.